smithy-utils/src/main/java/software/amazon/smithy/utils/IoUtils.java

```diff
@@ -365,6 +368,40 @@ public FileVisitResult postVisitDirectory(Path dir, IOException e) throws IOExce
         return true;
     }
 
+    /**
+     * Computes the sha1 digest of a file.
+     *
+     * @param path Path to file to compute hash for.
+     * @return sha1 digest string.
+     * @throws UncheckedIOException if the specified file could not be read.
+     */
+    public static String computeSha1(Path path) {
```

Comment:
I don't know that this is a generalizable problem at this point that warrants this kind of global utility function. I'd recommend keeping this internal to the dependency management stuff for now.